core/identity-hub-core/src/main/java/org/eclipse/edc/identityhub/DefaultServicesExtension.java

```diff
@@ -105,4 +111,11 @@ public RevocationListService createRevocationListService(ServiceExtensionContext
     public SignatureSuiteRegistry createSignatureSuiteRegistry() {
         return new InMemorySignatureSuiteRegistry();
     }
+
+    @Provider(isDefault = true)
+    public JwsSignerProvider defaultSignerProvider() {
+        // default implementation: resolve the private key (from vault of config) and create a JWSSigner based on its algorithm
+        return privateKeyId -> privateKeyResolver.resolvePrivateKey(privateKeyId)
```

Comment:
this is also duplicated in the connector, maybe it could be extracted in a class and put in the `crypto-common-lib` module

Reply:
need to merge #4403 first